packages/element/src/react-platform.native.js

```diff
@@ -8,27 +8,39 @@ import { omit } from 'lodash';
  * WordPress dependencies
  */
 import { applyFilters, doAction } from '@wordpress/hooks';
+import { Component } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { cloneElement } from './react';
 
-const render = ( element, id ) =>
-	AppRegistry.registerComponent( id, () => ( propsFromParent ) => {
-		const parentProps = omit( propsFromParent || {}, [ 'rootTag' ] );
+const render = ( element, id ) => {
+	class App extends Component {
```

Comment:
The `App` class component is defined within the `render` function because it uses the `element` argument to render the component.